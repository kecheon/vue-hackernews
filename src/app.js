import Vue from 'vue'
import Vuex from 'vuex'
import Router from 'vue-router'
import { sync } from 'vuex-router-sync'
import App from './App.vue'
import storeConfig from './store/store-config'
import routerConfig from './router/router-config'
import {titleMixin} from './util/mixins'
import {
  timeAgo,
  host
} from './util/filters'

// mixin for handling title
Vue.mixin(titleMixin)

// register global utility filters.
Vue.filter('timeAgo', timeAgo)
Vue.filter('host', host)

// Expose a factory function that creates a fresh set of store, router,
// app instances on each call (which is called for each SSR request)
export function createApp () {
  // create store and router instances
  Vue.use(Vuex)
  Vue.use(Router) // #B

  const router = new Router(routerConfig)
  const store = new Vuex.Store(storeConfig)

  // sync the router with the vuex store.
  // this registers `store.state.route`
  sync(store, router)

  // create the app instance.
  // here we inject the router, store and ssr context to all child components,
  // making them available everywhere as `this.$router` and `this.$store`.
  const app = new Vue({
    router,
    store,
    render: h => h(App)
  })

  // expose the app, the router and the store.
  // note we are not mounting the app here, since bootstrapping will be
  // different depending on whether we are in a browser or on the server.
  return { app, router, store }
}
