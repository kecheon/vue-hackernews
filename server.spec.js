import request from 'supertest' // #A
import app from './server'

test('/top returns 200', () => {
  return request(app) // #B
    .get('/top') // #C
    .expect(200)
})

test('returns a 404 when page does not exist', () => {
  return request(app)
    .get('/does-not-exist')
    .expect(404)
    .expect(res => {
      expect(res.text).toMatchSnapshot()
    })
})
