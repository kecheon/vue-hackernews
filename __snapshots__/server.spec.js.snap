// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a 404 when page does not exist 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\">
    <title>vue-hackernews-starter</title>
  <link rel=\\"preload\\" href=\\"/dist/manifest.692a165441e41370b206.js\\" as=\\"script\\"><link rel=\\"preload\\" href=\\"/dist/vendor.ef6775f4539cdb66e821.js\\" as=\\"script\\"><link rel=\\"preload\\" href=\\"/dist/app.a4675157dc5139f20b58.js\\" as=\\"script\\"><link rel=\\"preload\\" href=\\"/dist/common.a4675157dc5139f20b58.css\\" as=\\"style\\"><link rel=\\"stylesheet\\" href=\\"/dist/common.a4675157dc5139f20b58.css\\"></head>
  <body>
    <div id=\\"app\\" data-server-rendered=\\"true\\"><header class=\\"header\\"><nav class=\\"inner\\"><a href=\\"/top\\">Top</a> <a href=\\"/new\\">New</a> <a href=\\"/show\\">Show</a> <a href=\\"/ask\\">Ask</a> <a href=\\"/job\\">Jobs</a></nav></header> <div class=\\"view\\"><div class=\\"container\\"><h1>404</h1> <p>The page you're looking for could not be found. Try selecting one of the lists in the nav bar.</p></div></div></div><script src=\\"/dist/manifest.692a165441e41370b206.js\\" defer></script><script src=\\"/dist/vendor.ef6775f4539cdb66e821.js\\" defer></script><script src=\\"/dist/app.a4675157dc5139f20b58.js\\" defer></script>
  </body>
</html>
"
`;
